Amazon.Ion.ObjectMapper/IonListSerializer.cs

```diff
@@ -1,32 +1,57 @@
 using System;
+using System.Collections;
 using System.Collections.Generic;
 using Amazon.IonDotnet;
 
 namespace Amazon.Ion.ObjectMapper
 {
-    public class IonListSerializer : IonSerializer<System.Collections.IList>
+    public class IonListSerializer : IonSerializer<IList>
     {
         private readonly IonSerializer serializer;
-        private readonly Type listType;
-        private readonly Type elementType;
-        private readonly bool isGenericList;
+        private Type listType;
+        private Type elementType;
+        private bool isGenericList;
 
-        public IonListSerializer(IonSerializer serializer, Type listType, Type elementType)
+        public IonListSerializer(IonSerializer serializer)
```

Comment:
Why has this changed? How is this related to the custom serialization change?

Reply:
This was done because we were allowing the IonSerializer to change the type but we are no longer doing that.